### PR TITLE
Pantry debug

### LINKED ISSUE #38: Fix - Negative Recipe Costs
Some recipes are returning negative..... WHY!

### PR DESCRIPTION
**Pull request by kellydinneen**
* Is this a fix or a feature?
fix

* What is the change/fix?
fixes bug in calculation of missing ingredient cost
displays info using domUpdate function

* Where should the reviewer start?
pantry.js and scripts.js

* How should this be tested?
using pantry test file and by opening webpage

### CODE REVIEW

**Comment by kellydinneen** [comment]
Info is displaying without error messages, but I noticed that all cards are saying user DOESN'T have enough to cook. Not sure if this is coincidence of data or problem with our function.

**Comment by kellydinneen** [comment]
Oh! Just realized the recipe we return with `getMissingPartOfRecipe(recipe)` includes ingredients that we need 0 of, but this means missing ingredients list is super long. Plus, its unnecessary for calculating total cost because 0 of something doesn't cost anything. Will fix

**Comment by kellydinneen** [comment]
Newest commit makes sure ONLY missing ingredients are displayed: I'm thinking data may just be such that no user has all ingredients for any recipe...kind of an issue if we want to show off our 'cook' button display.

File: src/scripts.js [hunk 1]
```diff
@@ -71,11 +71,14 @@ function onStartup() {
...
 function compilePantryData(recipesList) {
   recipesList.forEach(recipe => {
-    let missingIngredients = pantry.getMissingPartOfRecipe(recipe).ingredients.map(ingredient => {
-      let specificIngredient = ingredientData.find(item => item.id === ingredient.id);
-      return specificIngredient.name;
-    });
-
+    let missingIngredients = pantry.getMissingPartOfRecipe(recipe).ingredients.reduce((acc, ingredient) => {
+      if(ingredient.quantity.amount > 0) {
+        let specificIngredient = ingredientData.find(item => item.id === ingredient.id);
```
Comment (kellydinneen) [inline src/scripts.js:76]: lines 76 - 77: find name of ingredient (not included in recipe) in ingredientsData

File: src/scripts.js [hunk 2]
```diff
@@ -71,11 +71,14 @@ function onStartup() {
 
 function compilePantryData(recipesList) {
   recipesList.forEach(recipe => {
-    let missingIngredients = pantry.getMissingPartOfRecipe(recipe).ingredients.map(ingredient => {
-      let specificIngredient = ingredientData.find(item => item.id === ingredient.id);
-      return specificIngredient.name;
-    });
-
+    let missingIngredients = pantry.getMissingPartOfRecipe(recipe).ingredients.reduce((acc, ingredient) => {
+      if(ingredient.quantity.amount > 0) {
```
Comment (kellydinneen) [inline src/scripts.js:75]: line 75 checks to make sure we don't have all that we need of the ingredient (we could exclude these ingredients earlier within the `getMissingPartOfRecipe()` function, but we'd have to change our `.map` to something else)

File: src/scripts.js (hunk 1 above)
Comment (kevinhartmann23) [reply src/scripts.js:76]: I noticed that this morning too, could be a new method under recipe class to add name to each ingredient, simple to how we are adding estimated cost!

**Review by kelsiebesingeryeh — APPROVED** [review]
Looks good

**Review by kevinhartmann23 — APPROVED** [review]
Looks great, all ideas on where to move forward make sense.

**Pull request merged**